### PR TITLE
Add use ref

### PR DESCRIPTION
**Pull request by Cwarcup**
```jsx
  const loggedInUserRef = useRef(null); // initialize ref to null, no user yet
```
> used to setup `useRef`

 
function to set UserRef and user state. gets called inside of the axios request if server returns status code 200 (OK).
```jsx
    const successfullLogin = (response) => {
      loggedInUserRef.current = response.data;
      setUser(loggedInUserRef.current); // ?? note sure if we actually need this.
    };
```

We have two return statements. 

1. If the `successfullLogin` gets run, then we got status code 200, meaning user is in teh database. 
```jsx
  if (loggedInUserRef.current) {
    const { name, email } = loggedInUserRef.current;
    return (
      <div>
        <h1>Welcome {name}</h1>
        <h2>Your email is {email}</h2>
        <TypingField />
      </div>
    );
// * need to add correct component to render, props, etc.
```

If we do **not** get status code 200, the `Login` component just render the form again:
```jsx
// if status code != 200 -> go sad route
  return (
    <Fragment>
      <form
        onSubmit={(event) => {
          event.preventDefault();
        }}
        autoComplete="off"
      >
        <input
          value={email}
          type="text"
          placeholder="Email"
          onChange={(event) => {
            setEmail(event.target.value);
          }}
        />
      </form>
// ... and so on
}
```

### CODE REVIEW

**Comment by Cwarcup** [comment]
Also made small changes to the server

```rb
class SessionsController < ApplicationController
  def create
    if (@user = User.find_by(email: params[:session][:email].strip.downcase))
      if @user.email == params[:session][:email] && @user.password == params[:session][:password]
        @user = User.find_by(email: params[:session][:email].strip.downcase)
        # if the user exists AND the password entered is correct
        render json: @user, status: 200
      end
    else
      # if user's login doesn't work, send them back to the login form
      render json: 'There was something wrong with your login details.', status: 401
    end
  end

  def destroy
    session[:user_id] = nil
  end
end
```

File: src/components/Login.jsx [hunk 1]
```diff
@@ -1,74 +1,107 @@
...
+        .post(`http://localhost:3000/login`, loginData, {
+          headers: {
+            'content-type': 'application/json',
+          },
+        })
+        .then((res) => {
+          setUser({ name: res.data.name });
+          // if server returns 200 (success)
+          if (res.status === 200) {
+            successfullLogin(res);
```
Comment (maxwellsquared) [inline src/components/Login.jsx:38]: Make sure to change "successfullLogin" to "successfulLogin" everywhere!

File: src/components/Login.jsx [hunk 2]
```diff
@@ -1,74 +1,107 @@
...
+  const loggedInUserRef = useRef(null); // initialize ref to null, no user yet
+
+  function sendRequest(email, password, user, setUser) {
+    let loginData = {
+      email: email,
+      password: password,
+    };
+
+    // function to set UserRef and user state
+    const successfullLogin = (response) => {
```
Comment (maxwellsquared) [inline src/components/Login.jsx:22]: successfulLogin (two Ls, not three)

**Review by maxwellsquared — COMMENTED** [review]
note on changing "successfullLogin" [sic] to "successfulLogin"

**Pull request merged**